replayserver/bookkeeping/storage.py

```diff
@@ -92,8 +94,7 @@ def _write_replay(self, rfile, info, data):
         try:
             rfile.write(json.dumps(info).encode('UTF-8'))
             rfile.write(b"\n")
-            data = struct.pack(">i", len(data)) + zlib.compress(data)
-            data = base64.b64encode(data)
+            data = self._compressor.compress(data)
```

Comment:
Why are we removing the data length? It is useful for verifying that the replay file isn't corrupted as well as for being able to pre allocate your output buffer.

Reply:
Well, that's what I chose to do, I didn't know if that's anyhow useful. IIRC the zstd decompressor from apache commons doesn't need that target length. If the replay is corrupt, we'll probably get a decompression error anyway.

Reply:
Also, uncompressed size can be part of the zstd archive itself. The python library defaults to adding uncompressed size info, so we should be good.

Reply:
Is it part of the zstd data? Because I use this info in my parser, I would like if it's still available.

Reply:
It's in archive's headers, so I guess it's accessible. If you're talking about the parser that's run by the replay server, it runs the parser before compressing the data, doesn't it? We can query uncompressed data size there easily. Otherwise I suppose you'll just have to rely on whatever decompresser you're using.

Reply:
The replay server only uses the parts that operate on the raw scfa data, but I have also implemented the FAF format for convenience.

Reply:
Cool, I just looked it up. The C implementation has a function `ZSTD_getFrameContentSize()` which returns the uncompressed size. But according to the documentation, this field is optional and might not be present if zstd is used in stream mode. Are we using stream mode? I would think it doesn't make sense for our use case.

Reply:
We're not. Technically we *could*, in a desperate bid to conserve RAM usage, but that's not needed for now.